### PR TITLE
Remove ChronosInterface references and clean up ChronosDate and ChronosTime refs

### PR DESCRIPTION
**Pull request by othercorey**
Removes ChronosInterface and cleans up list of classes provided by Chronos.

The docs need more information added, but this at least makes them correct.

### CODE REVIEW

File: docs/en/index.rst [hunk 1]
```diff
@@ -18,22 +26,6 @@ following::
 
     php composer.phar require "cakephp/chronos:^3.0"
 
-Overview
```
Comment (othercorey) [inline docs/en/index.rst:21]: After fixing both the intro and this section, it felt like a duplicate. Merged into the intro at the top.

File: docs/en/index.rst [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 
-Chronos provides a zero-dependency collection of extensions to the ``DateTime``
-object. In addition to convenience methods, Chronos provides:
+Chronos provides a zero-dependency set of ``DateTimeImmutable`` extension, Date-only and Time-only classes:
 
-* ``Date`` objects for representing calendar dates.
-* Immutable date and datetime objects.
+* ``Cake\Chronos\Chronos`` extends ``DateTimeImmutable`` and provides many helpers.
+* ``Cake\Chronos\ChronosDate`` represents calendar dates unaffected by time or time zones.
+* ``Cake\Chronos\ChronosTime`` repreents clock times independent of date or time zones.
```
Comment (ADmad) [inline docs/en/index.rst]:
```suggestion
* ``Cake\Chronos\ChronosTime`` represents clock times independent of date or time zones.
```

File: docs/en/index.rst [hunk 3]
```diff
@@ -1,14 +1,22 @@
 Chronos
 #######
 
-Chronos provides a zero-dependency collection of extensions to the ``DateTime``
-object. In addition to convenience methods, Chronos provides:
+Chronos provides a zero-dependency set of ``DateTimeImmutable`` extension, Date-only and Time-only classes:
```
Comment (markstory) [inline docs/en/index.rst]:
```suggestion
Chronos provides a zero-dependency ``DateTimeImmutable`` extension, Date-only and Time-only classes:
```

File: docs/en/index.rst [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
+* ``Cake\Chronos\ChronosDate`` represents calendar dates unaffected by time or time zones.
+* ``Cake\Chronos\ChronosTime`` represents clock times independent of date or time zones.
+* Only safe, immutable objects.
 * A pluggable translation system. Only English translations are included in the
   library. However, ``cakephp/i18n`` can be used for full language support.
 
+The ``Chronos`` class extends ``DateTimeImmutable`` and implements ``DateTimeInterface``
+which allows users to use type declarations that support either.
+
+ ``ChronoDate`` and ``ChronosTime`` do not extend ``DateTimeImmutable`` and do not
```
Comment (markstory) [inline docs/en/index.rst]:
```suggestion
 ``ChronosDate`` and ``ChronosTime`` do not extend ``DateTimeImmutable`` and do not
```

**Pull request merged**